solidity/app/contracts/Router.sol

```diff
@@ -130,22 +132,50 @@ abstract contract Router is AbacusConnectionClient, IMessageRecipient {
     }
 
     /**
-     * @notice Dispatches a message to an enrolled router via the local router's
-     * Outbox and pays for message processing on the destination chain.
-     * @dev Reverts if there is no enrolled router for _destination.
+     * @notice Pay for message processing on the destination
+     * @param _leafIndex The leaf index of the message to pay processing for
+     * @param _gasPayment The amount of native tokens to pay the Interchain Gas
+     * Paymaster to process the dispatched message.
+     */
+    function _payForGas(uint256 _leafIndex, uint256 _gasPayment) internal {
```

Comment:
Why `payForGas` when the function on the paymaster is `payGasFor`? Isn't that confusing?

Reply:
Indeed